source/extensions/filters/network/http_connection_manager/config.cc

```diff
@@ -298,6 +299,25 @@ HttpConnectionManagerConfig::HttpConnectionManagerConfig(
   }
   request_id_extension_ = Http::RequestIDExtensionFactory::fromProto(final_rid_config, context_);
 
+  // Check if IP detection extensions were configured.
+  const auto& ip_detection_extensions = config.original_ip_detection_extensions();
+  if (!ip_detection_extensions.empty()) {
+    original_ip_detection_extensions_.reserve(ip_detection_extensions.size());
+    for (const auto& extension_config : ip_detection_extensions) {
+      auto* factory =
+          Envoy::Config::Utility::getFactory<Http::OriginalIPDetectionFactory>(extension_config);
+      if (!factory) {
+        throw EnvoyException("Original IP detection extension not found");
```

Comment:
will this print out the name of the missing extension?  Might be helpful for debug.